web/src/settings_emoji.ts

```diff
@@ -111,10 +111,16 @@ export function populate_emoji(): void {
     const emoji_data = emoji.get_server_realm_emoji_data();
     const active_emoji_data = Object.values(emoji_data).filter((emoji) => !emoji.deactivated);
 
+    const ignore_missing_authors = !settings_data.user_can_access_all_other_users();
     for (const emoji of active_emoji_data) {
         // Add people.js data for the user here.
         if (emoji.author_id !== null) {
-            emoji.author = people.maybe_get_user_by_id(emoji.author_id);
+            const author = people.maybe_get_user_by_id(emoji.author_id, ignore_missing_authors);
+            if (author === undefined) {
+                emoji.author = {full_name: people.INACCESSIBLE_USER_NAME};
+            } else {
+                emoji.author = {full_name: author.full_name};
+            }
```

Comment:
This might break live-update if an author's name changes for a real user. 

I feel like we could just generate a fake user object using `make_user` to point to instead of having custom logic in this code path?